tests/unit/utils/test_txn_parser.py

```diff
@@ -0,0 +1,454 @@
+                    },
+                    "LedgerEntryType": "AccountRoot",
+                    "LedgerIndex": "8719BD164C9F79A760E19FB64691885E63CC595032E559971F8"
+                    "9C1E22EAEC220",
+                    "PreviousFields": {
+                        "Balance": "2743653331433",
+                        "Sequence": 80949,
+                    },
+                    "PreviousTxnID": "8DD9649A8E789257A61CD7223A507602CED9F08BCF1FFC9F2"
+                    "E79C08DCC2863E8",
+                    "PreviousTxnLgrSeq": 69457007,
+                }
+            },
+            {
+                "ModifiedNode": {
+                    "FinalFields": {
+                        "Account": "rLQUo6VbTMHA7dFTpCTzSzTKAUuWSy7u3K",
+                        "Balance": "815244743639",
+                        "Flags": 131072,
+                        "OwnerCount": 0,
+                        "Sequence": 14175,
+                    },
+                    "LedgerEntryType": "AccountRoot",
+                    "LedgerIndex": "969F1B7A63CC54C344B2990659AFD3F8BB541D03B574B48EED4"
+                    "1A32DA51AA53A",
+                    "PreviousFields": {"Balance": "815032870725"},
+                    "PreviousTxnID": "150DB308EEC5BBB4037E8953D3D5B42371D49033012D16DC1"
+                    "170036DD76D2AD2",
+                    "PreviousTxnLgrSeq": 69456777,
+                },
+            },
+        ],
+        "TransactionIndex": 7,
+        "TransactionResult": "tesSUCCESS",
+        "delivered_amount": "211872914",
+    },
+    "validated": True,
+}
+
+
+class TestBalanceChanges(TestCase):
+    def test_valid_metadata_missing_account(self):
+        bc_test_no_account = copy.deepcopy(TEST_TXN)
+        bc_test_no_account["transaction"].pop("Account")
+        with self.assertRaises(xrpl.utils.XRPLTxnFieldsException):
+            xrpl.utils.parse_balance_changes(transaction=bc_test_no_account)
+
+    def test_valid_metadata_missing_meta(self):
+        bc_test_no_meta = copy.deepcopy(TEST_TXN)
+        bc_test_no_meta.pop("meta")
+        with self.assertRaises(xrpl.utils.XRPLTxnFieldsException):
+            xrpl.utils.parse_balance_changes(transaction=bc_test_no_meta)
+
+    def test_valid_metadata_missing_nodes(self):
+        bc_test_no_nodes = copy.deepcopy(TEST_TXN)
+        bc_test_no_nodes["meta"].pop("AffectedNodes")
+        with self.assertRaises(xrpl.utils.XRPLTxnFieldsException):
+            xrpl.utils.parse_balance_changes(transaction=bc_test_no_nodes)
+
+    def test_parse_balace_changes(self):
+        actual = xrpl.utils.parse_balance_changes(transaction=TEST_TXN)
+        expected = {
+            "rpXhhWmCvDwkzNtRbm7mmD1vZqdfatQNEe": [
+                {
+                    "Counterparty": "rhub8VRN55s94qWKDv6jmDy1pUykJzF3wq",
+                    "Currency": "USD",
+                    "Value": "-0.7042071790427",
+                },
+                {"Counterparty": "", "Currency": "XRP", "Value": "1.000000"},
```

Comment:
We should keep the standard XRPL parlance, to avoid confusion - so `issuer` instead of `Counterparty`. It would also be more Pythonic/fit the library better if it's all lowercase instead of being capitalized.

Reply:
ditto with all of the other methods as well

Reply:
If you parse balance changes for example, it returns the balance changes of all parties affected.
For example rPu2feBaViWGmWJhvaF5yLocTVD8FUxd2A bought 8.455426113856 CNY from RippleQK. The balance change for RippleQK looks like:
`'rPT74sUcTBTQhkHVD54WGncoqXEAMYbmH7': [{'Counterparty': 'rafjNPiCrvVniSGSj9UUpAPTvGnHsbeuKn', 'Currency': 'CNY', 'Value': '8.455426113856'}, {'Counterparty': 'rPu2feBaViWGmWJhvaF5yLocTVD8FUxd2A', 'Currency': 'CNY', 'Value': '-8.455426113856'}]}`
The counterparty for RippleQK is rPu2feBaViWGmWJhvaF5yLocTVD8FUxd2A but the account is not the issuer.
I think counterparty would be the better term :) I guess for the order book changes it would make sense.
It should be lowercase.